parsing/parser.mly

```diff
@@ -67,8 +67,20 @@ let mkpatvar name pos =
   it must be ghost.
 *)
 let ghexp d = Exp.mk ~loc:(symbol_gloc ()) d
-let ghpat d = Pat.mk ~loc:(symbol_gloc ()) d
-let ghtyp d = Typ.mk ~loc:(symbol_gloc ()) d
+let ghpat ?loc d =
+  let loc =
+    match loc with
+    | None -> symbol_gloc ()
+    | Some l -> { l with loc_ghost = true }
+  in
+  Pat.mk ~loc d
+let ghtyp ?loc d =
+  let loc =
+    match loc with
+    | None -> symbol_gloc ()
+    | Some l -> { l with loc_ghost = true }
+  in
```

Comment:
Can you factorize this match into a `ghost ?loc ()` function?

Reply:
Will do.